staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -349,8 +358,15 @@ func (g *Cloud) ensureInternalLoadBalancerDeleted(clusterName, clusterID string,
 		return err
 	}
 
+	klog.V(2).Infof("ensureInternalLoadBalancerDeleted(%v): Removing forwarding rule annotation on service %s", loadBalancerName, svcNamespacedName)
+	patchedSvc, err := patchILBNameAnnotation(g.client, svc, nil, "")
```

Comment:
Can we do this after removing the finalizer, in line 373? Otherwise, it will cause one more service update to be queued due to the annotations change? - https://github.com/kubernetes/kubernetes/blob/a9e2e18cc6737072fef2bce838bcf508a8e708f3/staging/src/k8s.io/cloud-provider/controllers/service/controller.go#L525

Reply:
Since `removeFinalizer` is only used here, I will repurpose it to remove the finalizer and the service annotation in one patch update. If we do it after removing the finalizer, we may enter into a race of the resource being deleted and trying to patch the resource.